src/main/java/pl/javastart/couponcalc/PriceCalculator.java

```diff
@@ -1,13 +1,55 @@
 package pl.javastart.couponcalc;
 
-import java.util.List;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+import java.util.stream.Collectors;
 
 public class PriceCalculator {
 
     public double calculatePrice(List<Product> products, List<Coupon> coupons) {
-        // TODO
+        if (Objects.isNull(products) && Objects.isNull(coupons)) {
+            return 0;
+        }
 
-        return 0;
+        if (Objects.isNull(coupons)) {
+            return products.stream()
+                    .mapToDouble(Product::getPrice)
+                    .sum();
```

Comment:
Liczysz sumę cen produktów tutaj i później jeszcze raz, ale w inny sposób, można to zrobić tylko raz